doc/sonic-build-system/web_file_server_population_script.md

```diff
@@ -0,0 +1,127 @@
+
+### In scope:
+- Download web packages from external file storage
+- Rename and upload web packages to trusted file storage
+- Script output
+
+## Download web packages from external file storage
+The script will identify all relevant web packages, collect their hash value and url and then download them from the remote site to a local cache.
+The script search over the source tree path recursively and look for ‘version-web’ files.
+Those file are being parsed in order to retrieve the path to the package and its hash value. The script is expecting for such format:
+[package url]==[hash value]
+
+```sh
+For example:
+https://storage.googleapis.com/golang/go1.14.2.linux-amd64.tar.gz==ebef065e4d35572af5b03e2be957a9c6c5063b38
+```
+
+### Work Items:
+- Identify relevant versions-web files
+- Parse versions-web files and identify relevant web packages
+- Download web packages from their external file storage
+
+
+## Upload web packages to trusted file storage
+
+Although there is no version for a package downloaded by wget/curl, the hash value of the package file can be used as the version value. 
+The script would rename the Web package and then upload it from cache to a trusted file server. The file name format in the file storage is as below:
+[original file name]-[hash value]
+```sh
+For an example: 
+go1.14.2.linux-amd64.tar.gz-ebef065e4d35572af5b03e2be957a9c6c5063b38
+```
+For the same web package, it supports to have multiple versions of the file in the file storage
+for example:
+```sh
+For an example: 
+go1.14.2.linux-amd64.tar.gz-ebef065e4d35572af5b03e2be957a9c6c5063b38
+go1.14.2.linux-amd64.tar.gz-b73d6366c0bce441d20138c150e2ddcdf406e654 (fake hash value)
+```
+Same file name is used with different hash 
+
+### Work Items:
+- Rename web package
+- Upload web package to trusted file storage
+- Generate output version file (optional)
+
+## User Input:
+The user would provide the script the following parameters:
+
+| Name        	| Description                                                                                                	| Tag          	| Type    	| Default value                      	|
+|-------------	|------------------------------------------------------------------------------------------------------------	|--------------	|---------	|------------------------------------	|
+| source      	| Search path for source tree where to  look for<br> compilation artifacts (version files)                   	| -s, --source 	| String  	| CWD<br>(current working directory) 	|
+| cache       	| Path for temporary storage where to download Web Packages                                                  	| -c, --cache  	| String  	| CWD/tmp                            	|
+| output      	| Output file name to hold the list of packages.<br> The file will be created in the format of version files 	| -o, --output 	| String  	| none                               	|

```

Comment:
Can you please provide an example of version files?

Reply:
We have an example of the format of version file in line 48. Do you think we need to add additional examples?